### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by LaskoA**
Check this out :)

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,68 @@
-# write your code here
+class SoftwareEngineer:
+    skills = []
```
Comment (Abnormaltype) [inline app/main.py:2]: `skills` - should be an instance attribute

File: app/main.py [hunk 2]
```diff
@@ -1 +1,68 @@
...
+    skills = ["Python", "SQL", "Django"]
+
+    def __init__(self, name):
+        super().__init__(name)
+
+    def create_powerful_api(self):
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+    def learn_skill(self, skill):
```
Comment (Abnormaltype) [inline app/main.py:35]: same here

File: app/main.py [hunk 3]
```diff
@@ -1 +1,68 @@
...
+    def create_awesome_web_page(self):
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+    def learn_skill(self, skill):
+        FrontendDeveloper.skills.append(skill)
+
+
+class BackendDeveloper(SoftwareEngineer):
+    skills = ["Python", "SQL", "Django"]
```
Comment (Abnormaltype) [inline app/main.py:26]: same here

**Review by Abnormaltype — CHANGES_REQUESTED** [review]
fix in all classes

**Comment by LaskoA** [comment]
Please check now (I was to self-confident previous time)

File: app/main.py [hunk 4]
```diff
@@ -1 +1,55 @@
...
+                         SoftwareEngineer):
+
+    def __init__(self, name):
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django", "JavaScript", "HTML", "CSS"]
+
+    def create_web_application(self):
+        print(f"{self.name} started creating a web application...")
+        BackendDeveloper.create_powerful_api(self)
+        FrontendDeveloper.create_awesome_web_page(self)
```
Comment (Abnormaltype) [inline app/main.py:55]: use inheritance

File: app/main.py [hunk 5]
```diff
@@ -1 +1,55 @@
...
+        self.skills = ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self):
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(BackendDeveloper,
+                         FrontendDeveloper,
+                         SoftwareEngineer):
```
Comment (Abnormaltype) [inline app/main.py:46]: Do you need `SoftwareEngineer`?

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
Do you know that "3" is my lucky number?)

**Review by Abnormaltype — CHANGES_REQUESTED** [review]
3(

**Comment by LaskoA** [comment]
#4, I am sorry :(

**Review by Abnormaltype — APPROVED** [review]
Well done!)
Never give up)

**Pull request closed without merging**